server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -902,14 +911,13 @@ private static Mapper getLeafMapper(final DocumentParserContext context, String
         if (fieldType != null) {
             // we haven't found a mapper with this name above, which means if a field type is found it is for sure a runtime field.
             assert fieldType.hasDocValues() == false && fieldType.isAggregatable() && fieldType.isSearchable();
-            return NO_OP_FIELDMAPPER;
+            return noopFieldMapper(fieldPath);
         }
         return null;
     }
 
-    private static final FieldMapper NO_OP_FIELDMAPPER = new FieldMapper(
-        "no-op",
-        new MappedFieldType("no-op", false, false, false, TextSearchInfo.NONE, Collections.emptyMap()) {
+    private static FieldMapper noopFieldMapper(String path) {
```

Comment:
This is needed to correctly implement `fullPath()` that is now used in `context.isCopyToDestinationField` check.